### PR TITLE
Remove use of Cloudflare zlib

### PR DESCRIPTION
**Pull request by pugnascotia**
Relates to #81662. This library isn't ready for public consumption.
Remove it from the Docker build.

### CODE REVIEW

**Review by cburgess — COMMENTED** [review]
As the reporter of the original issue, I'm a fan of this approach. +1

File: distribution/docker/build.gradle [hunk 1]
```diff
@@ -302,8 +288,7 @@ void addTransformDockerContextTask(Architecture architecture, DockerBase base) {
       eachFile { FileCopyDetails details ->
         if (details.name.equals("Dockerfile")) {
           filter { String filename ->
-            String result = filename.replaceAll('^RUN curl.*artifacts-no-kpi.*$', "COPY ${distributionName} /tmp/elasticsearch.tar.gz")
-            return result.replaceAll('^RUN curl.*cloudflare/zlib.*$', "COPY zlib-${cloudflareZlibVersion}.tar.gz /tmp")
+            reeturn filename.replaceAll('^RUN curl.*artifacts-no-kpi.*$', "COPY ${distributionName} /tmp/elasticsearch.tar.gz")
```
Comment (wallrik) [inline distribution/docker/build.gradle:291]: Looks like a typo here. 😊
```suggestion
            return filename.replaceAll('^RUN curl.*artifacts-no-kpi.*$', "COPY ${distributionName} /tmp/elasticsearch.tar.gz")
```

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/packaging-upgrade-tests

**Comment by mark-vieira** [comment]
I think it's worth pinging @jpountz and @elastic/es-perf on this one.

**Comment by DJRickyB** [comment]
Thank you!

**Pull request merged**